scripts/annotate_vcf_bam.py

```diff
@@ -0,0 +1,203 @@
+                unique_coverage = 0.0
+                total_coverage = 0.0
+
+                num_forward = 0.0
+
+                bases_aligned = 0.0
+                total_bases = 0.0
+
+                end_bases_aligned = 0.0
+                end_total_bases = 0.0
+
+                num_discordant_high = 0.0
+                num_discordant_low = 0.0
+
+                num_repeat = 10
+                for i in xrange(0, num_repeat):
+                    loc = random.randint(breakpoints[0], breakpoints[1])
+                    alignments = sam_file.fetch(vcf_record.CHROM, loc, loc + 1)
+                    for rec in alignments:
+                        if rec.mapq >= 18:
+                            unique_coverage += 1
+                            if not rec.is_reverse:
+                                num_forward += 1
+                            total_bases += rec.rlen
+                            bases_aligned += rec.qlen
+
+                        total_coverage += 1
+
+                # compute number of discordant
+                for loc in [max(breakpoints[0] - sd_insert_size, 0), breakpoints[1] + sd_insert_size]:
+                    alignments = sam_file.fetch(vcf_record.CHROM, loc, loc + 1)
+                    for rec in alignments:
+                        if rec.mapq >= 18:
+                            if abs(rec.tlen) > template_upper_bound:
+                                num_discordant_high += 1
+                            if abs(rec.tlen) < template_lower_bound:
+                                num_discordant_low += 1
+                            end_total_bases += rec.rlen
+                            end_bases_aligned += rec.qlen
+
+                # get coverage between the breakpoints
+                vcf_record.INFO["AA_UNIQ_COV"] = (unique_coverage/num_repeat)/mean_coverage
+                vcf_record.INFO["AA_TOTAL_COV"] = (total_coverage/num_repeat)/mean_coverage
+
+                # get strand bias
+                if unique_coverage > 0.0:
+                    vcf_record.INFO["AA_TOTAL_STRAND"] = (num_forward/unique_coverage - 0.5) ** 2
+
+                # get mapping quality stats
+                if total_coverage > 0.0:
+                    vcf_record.INFO["AA_PROP_REPEAT"] = unique_coverage/total_coverage
+
+                # get clipped reads stats
+                if total_bases > 0.0:
+                    vcf_record.INFO["AA_PROP_ALIGNED"] = bases_aligned/total_bases
+                if end_total_bases > 0.0:
+                    vcf_record.INFO["AA_END_PROP_ALIGNED"] = end_bases_aligned/end_total_bases
+
+                # get discordant reads stats
+                vcf_record.INFO["AA_DISCORDANT_HIGH"] = num_discordant_high
+                vcf_record.INFO["AA_DISCORDANT_LOW"] = num_discordant_low
+
+                # get supplementary alignment stats
+                # Skip this for now
+
+            vcf_writer.write_record(vcf_record)
+
+        vcf_writer.close()
```

Comment:
you are right. what was i thinking. the iteration should take care of this.

On Thu, Feb 12, 2015 at 5:13 PM, John Mu <notifications@github.com> wrote:

> In scripts/annotate_vcf_bam.py
> <https://github.com/bioinform/metasv/pull/25#discussion_r24637294>:
>
> > +                # get clipped reads stats
> > +                if total_bases > 0.0:
> > +                    vcf_record.INFO["AA_PROP_ALIGNED"] = bases_aligned/total_bases
> > +                if end_total_bases > 0.0:
> > +                    vcf_record.INFO["AA_END_PROP_ALIGNED"] = end_bases_aligned/end_total_bases
> > +
> > +                # get discordant reads stats
> > +                vcf_record.INFO["AA_DISCORDANT_HIGH"] = num_discordant_high
> > +                vcf_record.INFO["AA_DISCORDANT_LOW"] = num_discordant_low
> > +
> > +                # get supplementary alignment stats
> > +                # Skip this for now
> > +
> > +            vcf_writer.write_record(vcf_record)
> > +
> > +        vcf_writer.close()
>
> it won't close :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bioinform/metasv/pull/25/files#r24637294>.
>



-- 

<http://www.bina.com>

*Marghoob Mohiyuddin, PhD  *//  Sr. Bioinformatics Scientist, Bina
Technologies
E. marghoob@bina.com

*Learn More*  //  video <http://www.bina.com/product.html>  //  news
<http://www.bina.com/press-coverage.html>  //  posters
<http://f1000.com/posters/search?query=bina+technologies>  //  science
<http://www.bina.com/science.html>  //  platform
<http://www.bina.com/platform.html>
*Test Drive Bina  //  Experience NextGen analysis for yourself *
<http://www.bina.com/test-drive.html>